pkg/instrumentation/auto/callback.go

```diff
@@ -60,38 +67,88 @@ func createPatch(obj client.Object) (client.Patch, error) {
 }
 
 func (m *AnnotationMutators) patchFunc(ctx context.Context, callback objectCallbackFunc) objectCallbackFunc {
-	return func(obj client.Object) bool {
+	return func(obj client.Object, _ any) (any, bool) {
 		patch, err := createPatch(obj)
 		if err != nil {
 			m.logger.Error(err, "Unable to create patch",
 				"kind", fmt.Sprintf("%T", obj),
 				"name", obj.GetName(),
 				"namespace", obj.GetNamespace(),
 			)
-			return false
+			return nil, false
 		}
-		if !callback(obj) {
-			return false
+		ret, ok := callback(obj, nil)
+		if !ok {
+			return ret, false
 		}
 		if err = m.clientWriter.Patch(ctx, obj, patch); err != nil {
 			m.logger.Error(err, "Unable to send patch",
 				"kind", fmt.Sprintf("%T", obj),
 				"name", obj.GetName(),
 				"namespace", obj.GetNamespace(),
 			)
-			return false
+			return ret, false
 		}
-		return true
+		return ret, true
 	}
 }
 
 func (m *AnnotationMutators) restartNamespaceFunc(ctx context.Context) objectCallbackFunc {
-	return func(obj client.Object) bool {
+	return func(obj client.Object, mutatedAnnotations_ any) (any, bool) {
```

Comment:
super minor nit: Consider renaming field. If it really can be anything, then at the moment of calling this function, you won't know if it's `mutatedAnnotations` or whatever. Maybe `previousResult` or `passToNext` like you have above. I would just try to keep it consistent.